sdk/security/keyvault/azadmin/backup/client_test.go

```diff
@@ -0,0 +1,151 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package backup_test
+
+import (
+	"context"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/recording"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azkeys"
+	"github.com/Azure/azure-sdk-for-go/sdk/security/keyvault/azadmin/backup"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBackupRestore(t *testing.T) {
+	client, sasToken := startBackupTest(t)
+
+	testSerde(t, &sasToken)
+
+	// backup the vault
+	backupPoller, err := client.BeginFullBackup(context.Background(), sasToken, nil)
+	require.NoError(t, err)
+	backupResults, err := backupPoller.PollUntilDone(context.Background(), nil)
+	require.NoError(t, err)
+	require.Nil(t, backupResults.Error)
+	require.Equal(t, "Succeeded", *backupResults.Status)
+	require.Contains(t, *backupResults.AzureStorageBlobContainerURI, os.Getenv("BLOB_RESOURCE_URI"))
+	testSerde(t, &backupResults)
+
+	// restore the backup
+	s := *backupResults.AzureStorageBlobContainerURI
+	folderURI := s[strings.LastIndex(s, "/")+1:]
+	restoreOperationParameters := backup.RestoreOperationParameters{
+		FolderToRestore:    &folderURI,
+		SasTokenParameters: &sasToken,
+	}
+	testSerde(t, &restoreOperationParameters)
+	restorePoller, err := client.BeginFullRestore(context.Background(), restoreOperationParameters, nil)
+	require.NoError(t, err)
+	restoreResults, err := restorePoller.PollUntilDone(context.Background(), nil)
+	require.NoError(t, err)
+	require.Nil(t, restoreResults.Error)
+	require.Equal(t, "Succeeded", *restoreResults.Status)
+	require.NotNil(t, restoreResults.StartTime)
+	require.NotNil(t, restoreResults.EndTime)
+	require.NotNil(t, restoreResults.JobID)
+	testSerde(t, &restoreResults)
+
+	// additional waiting to avoid conflicts with resources in other tests
+	if recording.GetRecordMode() != recording.PlaybackMode {
+		time.Sleep(61 * time.Second)
```

Comment:
I don't know a good alternative solution. It would be hard to condense the tests cases further, and without the wait time, it throws a "conflict" error. Borrowed the wait time solution from python: https://github.com/Azure/azure-sdk-for-python/blob/e8c3c96e035463a8c00daca5bae1644c3b7d5b43/sdk/keyvault/azure-keyvault-administration/tests/test_backup_client.py#L41

Reply:
leaving as is for now